ply-core/src/Ply/Core/Types.hs

```diff
@@ -87,25 +75,43 @@ data TypedScriptEnvelope = TypedScriptEnvelope
   }
   deriving stock (Eq, Show)
 
-instance FromJSON TypedScriptEnvelope' where
+newtype SerializedScript = SerializedScript {getSerializedScript :: ShortByteString}
+
+instance FromCBOR SerializedScript where
+  fromCBOR = SerializedScript . SBS.toShort <$> CBOR.fromCBOR
+
+cborToScript :: ByteString -> Either DecoderError Ledger.Script
+cborToScript x = deserialise . LBS.fromStrict . SBS.fromShort . getSerializedScript <$> CBOR.decodeFull' x
+
+instance FromJSON TypedScriptEnvelope where
   parseJSON (Object v) =
-    TypedScriptEnvelope'
+    TypedScriptEnvelope
       <$> v .: "version"
       <*> v .: "role"
       <*> v .: "params"
       <*> v .: "description"
-      <*> (parseJSONBase16 =<< v .: "cborHex")
-      <*> (parseJSONBase16 =<< v .: "rawHex")
+      <*> (parseAndDeserialize =<< v .: "cborHex")
```

Comment:
There's a subtle difference now because a json that didn't have a `rawHex` was previously invalid, but now it is valid. But that's probably fine.

Reply:
The previous version only checked if it was a valid Base16. It did not checked of `rawHex` matched `cborHex`. I don't think we are loosing anything here.

If you want I can make it read `rawHex` and check if both de-serialize into Script correctly--though, again, I don't really see we needing it.